### PR TITLE
Make Graph Marshaller a Widget

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage, @mlzummo Please Review

**Comment by mlzummo** [comment]
looks good, i see that you have another branch you are working on to incorp. the DDL_URL and the sub widgets in the discover section too

**Comment by mlzummo** [comment]
could you explain the PropertyEditor widget a bit more?

File: widgets/src/marshaller/Graph.js [hunk 1]
```diff
@@ -151,14 +151,44 @@
...
         this._dashboards = [];
         this.graphAttributes = ["snapToGrid", "showEdges"];
         this.widgetAttributes = ["layout", "chartType", "palette", "title", "columns", "data"];
     };
     Graph.prototype = Object.create(GraphWidget.prototype);
+    Graph.prototype.publish("ddl_url", "", "string", "DDL URL");
+    Graph.prototype.publish("proxy_mappings", [], "array", "Proxy Mappings");
+    Graph.prototype.publish("visualize_roxie", false, "boolean", "Show Roxie Data Sources");
+
+    Graph.prototype.publish()
```
Comment (GordonSmith) [inline widgets/src/marshaller/Graph.js]: Should not be here

**Comment by GordonSmith** [comment]
@mlzummo - The PropertyEditor widget is basically the html form from the dermatology page converted to a Widget.  Why?  We have widgets which contain other widgets (or lists of widgets) - this way we can now re-use the dermatology form in many places (and via recursion).

**Pull request merged**